### DISCUSSION TITLE
Proxmox API

### DISCUSSION THREAD

**Discussion by dbmello75** [opendcim/openDCIM#1571] [Q&A]
Hi, How can I get data from Proxmox???

**Comment by wilpig** [comment]
You need to create a user in proxmox that has access to the api, then on your host device in opendcim set the hypervisor to proxmox, give it the user/password and it will then be able to import.

**Reply by wilpig** [reply]
You can see it in action at https://dev.opendcim.org/devices.php?DeviceID=5070

**Comment by dbmello75** [comment]
I've created the user 
![image](https://github.com/user-attachments/assets/33b7ce6c-1130-485b-8d0e-ea93fe56918d)

Created the Token

![image](https://github.com/user-attachments/assets/76f5930e-7ebb-4c3e-a46b-da37d2d8a114)

Set it up in the OpenDCIM
![image](https://github.com/user-attachments/assets/e15e5622-c6d2-4f6d-972b-b75515cdf875)

Want I'm missing.

**Reply by wilpig** [reply]
Since I don't know what version you are running.  I have pushed this commit up for the current version that fixes a bug I believe is related to php8.  Past that here are the group assignments I have the api user in proxmox

https://github.com/opendcim/openDCIM/commit/e74f879a73b052df80f71480bc21f29d86ad2ad1

<img width="1058" alt="Screenshot 2024-10-25 at 2 19 32 PM" src="https://github.com/user-attachments/assets/e324f7b7-e100-406e-bba3-64f54915339a">
<img width="1294" alt="Screenshot 2024-10-25 at 2 20 03 PM" src="https://github.com/user-attachments/assets/7c3e3060-b885-467d-a759-53c1fab8abb6">

**Reply by wilpig** [reply]
<img width="633" alt="Screenshot 2024-10-25 at 2 21 22 PM" src="https://github.com/user-attachments/assets/713ad27c-4db6-44d8-90f3-51502f343cea">

**Comment by dbmello75** [comment]
I'm using openDCIM 23.04. Proxmox 8.2.2. 
I've created the group, the user. Still not working. Do I need to create an API Token????

I also noticed an error in the logs. 

/var/log/apache2/error.log

[Sat Oct 26 23:44:34.986876 2024] [php:error] [pid 10476:tid 10476] [client 192.168.88.165:53616] PHP Fatal error:  Uncaught ProxmoxVE\\Exception\\MalformedCredentialsException: PVE API needs a credentials object or an array. in /var/www/dcim/vendor/zzantares/proxmoxve/src/Credentials.php:65\nStack trace:\n#0 /var/www/dcim/classes/PMox.class.php(40): ProxmoxVE\\Credentials->__construct()\n#1 /var/www/dcim/classes/PMox.class.php(115): PMox::EnumerateVMs()\n#2 /var/www/dcim/devices.php(468): PMox::RefreshInventory()\n#3 {main}\n  thrown in /var/www/dcim/vendor/zzantares/proxmoxve/src/Credentials.php on line 65, referer: http://192.168.88.132/devices.php?DeviceID=2
